src/qibojit/custom_operators/backends.py

```diff
@@ -384,3 +383,252 @@ def swap_pieces(self, piece0, piece1, new_global, nlocal):
     def measure_frequencies(self, frequencies, probs, nshots, nqubits, seed=1234, nthreads=None):
         raise NotImplementedError("`measure_frequencies` method is not "
                                   "implemented for GPU.")
+
+
+class CuQuantumBackend(CupyBackend): # pragma: no cover
+    # CI does not test for GPU
+
+    def __init__(self):
+        super(CuQuantumBackend, self).__init__()
+        import cuquantum # pylint: disable=import-error
+        from cuquantum import custatevec as cusv # pylint: disable=import-error
+        self.cuquantum = cuquantum 
+        self.cusv = cusv
+        self.name = "cuquantum"
+
+    def get_cuda_type(self, dtype='complex64'):
+        if dtype == 'complex128':
+            return self.cuquantum.cudaDataType.CUDA_C_64F, self.cuquantum.ComputeType.COMPUTE_64F
+        elif dtype == 'complex64':
+            return self.cuquantum.cudaDataType.CUDA_C_32F, self.cuquantum.ComputeType.COMPUTE_32F
+        else:
+            raise TypeError("Type can be either complex64 or complex128")
+
+    def one_qubit_base(self, state, nqubits, target, kernel, qubits=None, gate=None):
+        # initialize cuStateVec library
+        handle = self.cusv.create()
+
+        ntarget = 1
+        target = nqubits - target - 1
+        target = self.np.asarray([target], dtype = self.np.int32)
+        ncontrols = len(qubits) - 1 if qubits is not None else 0
+        controls = self.np.asarray([i for i in qubits.get() if i != target], dtype = self.np.int32)
```

Comment:
I think this should be
```suggestion
        if qubits is not None:
            ncontrols = len(qubits) - 1
            controls = self.np.asarray([i for i in qubits.get() if i != target], dtype = self.np.int32)
        else:
            ncontrols = 0
            controls = self.np.empty() # or something like that ?
```
because as it is now it fails if `qubits` is None.

By the way this failure is not captured by tests, possibly because `qubits` is not None in any of the tests. I got it by playing with some other scripts.